packages/clerk-js/src/ui/utils/phoneUtils.ts

```diff
@@ -115,3 +116,31 @@ export function stringToFormattedPhoneString(str: string): string {
   const parsed = parsePhoneString(str);
   return `+${parsed.code} ${formatPhoneNumber(parsed.number, parsed.pattern, parsed.code)}`;
 }
+
+type CountryData = {
+  searchTerm: string;
+  country: CountryEntry;
+};
+
+const priorityCodes: string[] = ['1'];
+export function getLongestValidCountryCode(value: string, options: CountryData[]) {
+  let result = '';
+  let selectedCountryCode = null;
+  const numberValue = value.replace('+', '');
+
+  for (let i = 4; i > 0; i--) {
+    const code = numberValue.slice(0, i);
+    const priority = numberValue.slice(0, 1);
+    const hasPriority = priorityCodes.includes(priority);
```

Comment:
I think we should move this logic in the country [map itself](https://github.com/clerkinc/javascript/blob/2bbb13022e4a9d3cbd62f06ad2b06b229f05583e/packages/clerk-js/src/ui/elements/PhoneInput/countryCodeData.ts#L9) by adding a numeric priority as the fourth element in the countries that need it so the country map remains the single source of truth

You can then use the maps to check if a country has priority or not (and if both have, then prefer the country with the highest priority)